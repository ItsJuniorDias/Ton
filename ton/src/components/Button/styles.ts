import styled from 'styled-components/native';
import { scale } from '../../utils';

type DeleteProps = {
  isDelete: boolean | undefined;
};

export const Pressable = styled.Pressable<DeleteProps>`
  width: 100%;
  height: ${scale(26)};
  background-color: ${({ theme, isDelete }) =>
    isDelete ? theme.colors.alert : theme.colors.secondary};
  justify-content: center;
  border-radius: ${scale(16)};
  margin-top: ${scale(16)};
`;

export const Title = styled.Text<DeleteProps>`
  font-family: ${({ theme }) => theme.fonts.bold};
  font-size: ${scale(14)};
  color: ${({ theme, isDelete }) =>
    isDelete ? theme.colors.shape : theme.colors.text};
  text-align: center;
`;
