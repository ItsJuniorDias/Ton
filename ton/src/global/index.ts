export * from './shadowStyles';
export * from './theme';
