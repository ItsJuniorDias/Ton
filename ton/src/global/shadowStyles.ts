export const shadowStyles = {
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.1,
  shadowRadius: 2.62,
  elevation: 2,
};
