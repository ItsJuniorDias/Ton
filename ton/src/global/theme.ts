export const theme = {
  colors: {
    primary: '#01ad0b',
    secondary: '#01ea34',
    shape: '#ffffff',
    background: '#eff4f8',
    title: '#000000',
    text: '#20252b',
    notification: '#ff7b00',
    alert: '#ed5e69',
  },
  fonts: {
    regular: 'Roboto-Regular',
    medium: 'Roboto-Medium',
    bold: 'Roboto-Bold',
  },
};
