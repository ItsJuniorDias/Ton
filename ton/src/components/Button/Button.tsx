import React, { useRef } from 'react';
import { View } from 'react-native';

import { Pressable, Title } from './styles';

type ButtonProps = {
  title: string;
};

export const Button = ({ title }: ButtonProps) => {
  const testIDs = useRef({
    button: 'button_testID',
  }).current;

  return (
    <Pressable
      testID={testIDs.button}
      style={({ pressed }) => [
        {
          opacity: pressed ? 0.6 : 1,
        },
      ]}
    >
      <Title>{title}</Title>
    </Pressable>
  );
};
