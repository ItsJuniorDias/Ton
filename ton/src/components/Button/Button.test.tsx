import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { Button } from './Button';
import { theme } from '../../global';

jest.mock('react-native-iphone-x-helper', () => ({
  isIphoneX: jest.fn(),
}));

describe('Behavior Button', () => {
  const screenRender = (
    <ThemeProvider theme={theme}>
      <Button title="Pedir agora" />
    </ThemeProvider>
  );

  it('render snapshot', () => {
    const result = render(screenRender).toJSON();

    expect(result).toMatchSnapshot();
  });
  it('should render button', () => {
    const { getByTestId } = render(screenRender);

    const button = getByTestId('button_testID');

    expect(button).toBeTruthy();
  });
});
