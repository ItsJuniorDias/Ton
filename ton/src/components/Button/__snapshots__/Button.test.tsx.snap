// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior Button render snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#01ea34",
        "borderRadius": 31,
        "height": 51,
        "justifyContent": "center",
        "marginTop": 31,
        "width": "100%",
      },
      {
        "opacity": 1,
      },
    ]
  }
  testID="button_testID"
>
  <Text
    style={
      [
        {
          "color": "#20252b",
          "fontFamily": "Roboto-Bold",
          "fontSize": 27,
          "textAlign": "center",
        },
      ]
    }
  >
    Pedir agora
  </Text>
</View>
`;
